adapters/base/src/main/java/org/apache/cassandra/sidecar/adapters/base/TokenRangeReplicaProvider.java

```diff
@@ -115,22 +121,51 @@ private List<TokenRangeReplicas> transformRangeMappings(Map<List<String>, List<S
                               .collect(toList());
     }
 
-    private Map<String, String> replicaToStateMap(List<TokenRangeReplicas> replicaSet, StorageJmxOperations storage)
+    private List<ReplicaMetadata> getReplicaMetadata(List<TokenRangeReplicas> replicaSet,
+                                                     StorageJmxOperations storage,
+                                                     Map<String, String> hostToDatacenter)
     {
         List<String> joiningNodes = storage.getJoiningNodesWithPort();
         List<String> leavingNodes = storage.getLeavingNodesWithPort();
         List<String> movingNodes = storage.getMovingNodesWithPort();
 
+        List<String> liveNodes = storage.getLiveNodesWithPort();
+        List<String> deadNodes = storage.getUnreachableNodesWithPort();
+
+
         String rawGossipInfo = getRawGossipInfo();
         GossipInfoResponse gossipInfo = GossipInfoParser.parse(rawGossipInfo);
 
         StateWithReplacement state = new StateWithReplacement(joiningNodes, leavingNodes, movingNodes, gossipInfo);
+        RingProvider.Status status = new RingProvider.Status(liveNodes, deadNodes);
 
         return replicaSet.stream()
                          .map(TokenRangeReplicas::replicaSet)
                          .flatMap(Collection::stream)
                          .distinct()
-                         .collect(Collectors.toMap(Function.identity(), state::of));
+                         .map(r -> new ReplicaMetadata(state.of(r),
+                                                       status.of(r),
+                                                       resolveReplica(r),
+                                                       r,
+                                                       hostToDatacenter.get(r)))
+                         .collect(Collectors.toList());
+
+    }
+
+    private String resolveReplica(String r)
+    {
+        String hostName;
+        try
+        {
+            hostName = dnsResolver.reverseResolve(r.split(":")[0]);
+        }
+        catch (Exception e)
+        {
+            // Swallow exception to not fail-fast on DNS resolution failure
+            LOGGER.warn("Failed to resolve hostname for instance {}: {}", r, e);
+            hostName = "";
```

Comment:
why empty? maybe use the unresolved IP?

Reply:
I explicitly left this empty to denote the resolution failure, so the client can choose to handle it differently in this event. For instance, it can retry instead of making a sidecar request with the IP, which would fail to verify the client identity with TLS.

Reply:
that sounds like a bad idea. Why don't we mimic what `RingProvider` is doing instead. If there are host resolution errors, we should fail the request, client will see a 500, clients should leverage retries in that case.

Reply:
From offline discussion, the resolver will not fail on resolution failures. Decided to allow exception to be thrown on failures from the resolver. Addressed